scripts/daq_utils.py

```diff
@@ -0,0 +1,226 @@
+SLURM_PARTITION='drpq'
+SLURM_JOBNAME='submit_daq'
+DAQBATCH_SCRIPT='submit_daq.sh'
+DAQBATCH_OUTPUT='slurm_daq.log'
+MAX_RETRIES=10
+
+def silentremove(filename):
+    try:
+        os.remove(filename)
+    except OSError as e:
+        if e.errno != errno.ENOENT:  # errno.ENOENT = no such file or directory
+            raise
+
+def call_subprocess(*args):
+    cc = subprocess.run(args, stdout=PIPE, stderr=PIPE)
+    output = None
+    if not cc.returncode:
+        output = str(cc.stdout.strip(), "utf-8")
+    return output
+
+def call_sbatch(cmd, nodelist, scripts_dir):
+    sb_script = "#!/bin/bash\n"
+    sb_script += f"#SBATCH --partition={SLURM_PARTITION}" + "\n"
+    sb_script += f"#SBATCH --job-name={SLURM_JOBNAME}" + "\n"
+    sb_script += f"#SBATCH --nodelist={nodelist}" + "\n"
+    daqbatch_output = os.path.join(scripts_dir, DAQBATCH_OUTPUT)
+    sb_script += f"#SBATCH --output={daqbatch_output}" + "\n"
+    sb_script += f"#SBATCH --ntasks=1" + "\n"
+    sb_script += f"unset PYTHONPATH" + "\n"
+    sb_script += f"unset LD_LIBRARY_PATH" + "\n"
+    sb_script += cmd
+    daqbatch_script = os.path.join(scripts_dir, DAQBATCH_SCRIPT)
+    with open(daqbatch_script, "w") as f:
+        f.write(sb_script)
+    call_subprocess('sbatch', daqbatch_script)
+    silentremove(daqbatch_script)
+
+class SbatchManager:
+    def __init__(self, user):
+        self.user = user
+
+
+    def get_job_info(self):
+        # Use squeue to get JobId, Comment, JobName, Status, and Reasons (Node List)
+        format_string = '"%i %k %j %T %R"'
+        lines = call_subprocess(
+                    "squeue", "-u", self.user, "-h", "-o", format_string
+                ).splitlines()
+        job_details = {}
+        for i, job_info in enumerate(lines):
+            cols = job_info.strip('"').split()
+            success = True
+            if len(cols) == 5:
+                job_id, comment, job_name, state, nodelist = cols
+            elif len(cols) > 5:
+                job_id, comment, job_name, state = cols[:4]
+                nodelist = " ".join(cols[5:])
+            else:
+                success = False
+            if success:
+                # Get logfile from job_id
+                scontrol_lines = call_subprocess(
+                    "scontrol", "show", "job", job_id
+                ).splitlines()
+                logfile = ""
+                for scontrol_line in scontrol_lines:
+                    if scontrol_line.find("StdOut") > -1:
+                        scontrol_cols = scontrol_line.split("=")
+                        logfile = scontrol_cols[1]
+
+                job_details[job_name] = {
+                    "job_id": job_id,
+                    "comment": comment,
+                    "state": state,
+                    "nodelist": nodelist,
+                    "logfile": logfile,
+                }
+        return job_details
+
+
+class DaqManager:
+    def __init__(self, opts):
+        self.opts = opts
```

Comment:
I'm concerned about maintainability of scripts that pass CLI options through like this. Someone in the future is going to get confused and need to flip back and forth between this script and the CLI spec to understand what this is supposed to be. It's already kind of hard to follow as-written because there's no clear indication here of what self.opts might be. It'd be cleaner to pass through the different cli options as init args with full names, which also makes this simple to use outside of the CLI.

Something similar to:
```
def __init__(self, daq_host, verbose=False, some_other_option=False):
    self.daq_host = daq_host
    self.verbose = verbose
    self.some_other_option = some_other_option
```